content/docs/guides/continuous-delivery/troubleshooting-guide.md

```diff
@@ -0,0 +1,129 @@
+  * Note that if you do choose to use a pre-built binary, automatic plugin acquisition for Pulumi
+* You are caching the library dependencies but not the Pulumi plugins. Some services offer dependency caching by capturing a specific folder and restoring
+that folder whenever you run your pipeline subsequently. However, note that Pulumi dependencies have a post-install step that also pulls-down
+a [plugin]({{< relref "/docs/intro/concepts/how-pulumi-works#resource-providers" >}}) binary from our CDN.
+  * So be sure to cache the plugins path as well.
+  * If in doubt about problems encountered during execution, clear out all caches and restore dependencies from scratch.
+
+> Note that depending on the number of providers you use in your Pulumi app, the cache size can get big. Some CI services have
+> restrictions on the max size of the cache.
+
+## Cloud Provider Credentials
+
+Another common point of failure is the cloud provider credentials. The class of errors related to the credentials can be, but not limited to:
+
+* Incorrect credentials (wrong account, mismatched keys etc.)
+* Keys with a strict access scope that don't have access to creating specific resources
+
+### Tips
+
+* Cloud provider credentials not specified using a recognized mechanism. In almost all cases, specifying the credentials through environment variables will work.
+  * Some cloud providers also support authentication via their respective CLIs. For example, the Azure provider supports authentication via the Azure CLI.
+  * In the above example, if the Azure CLI is not properly authenticated, then Pulumi will not be able to acquire the necessary credentials.
+* Ensure that the Pulumi step has access to the environment variables containing the credentials.
+  * If your CI service has features that allow you to restrict which pipelines or steps in a pipeline can access the secrets, then ensure that
+  your pipeline/steps can access them.
+* Typo in the name of the environment variables. Refer to our [Cloud Providers]({{< relref "/docs/intro/cloud-providers" >}}) docs and the respective setup pages for each to find the correct
+environment variables to use.
+
+## Other Errors
+
+### Pulumi not found in the PATH
+
+This problem is especially prevalent for users using a Docker container-based pipeline. With most Docker container pipelines, the state of the container
+does not persist across steps, so you cannot split-out the installation step of the Pulumi CLI into another step and expect to use the modified environment
+in the following step. There are exceptions to this. Some CI services allow you to create a reusable template step that you can inject as a pre-cursor to other
+steps. This is a way to reduce repeating the installation step every time you would like to run `pulumi` command.  
+
+### Still need help?
+
+If your error is not related to any of the above categories, it is likely that there is a problem with the code in your infrastructure code.
```

Comment:
.. or it could be that we just didn't think of it or haven't documented it yet. ;) I do like referring to Slack for this, for sure. I'm not sure I'd point someone into the abyss of the programming model docs to troubleshoot CI problems, though. If they're here, they're likely beyond that level and looking for CI-specific info.

Reply:
Yes, very much agree with this. Reading this again, it actually doesn't sound nice at all. It seems to blame the user's code with high confidence. I will remove all of this and simply suggest to the user to reach out on our Community Slack.